app/controllers/api/v1/pets_controller.rb

```diff
@@ -10,12 +10,25 @@ def create
     end
   end
 
+  def index
+    pets = Pet.search(pet_type)
+    if pets
+      render json: PetSerializer.new(pets)
+    else
+      render json: { error: 'Pet type not recognized' }, status: :bad_request
+    end
+  end
+
   private
 
   def pet_params
     params.permit(:user_id, :name, :age, :pet_type, :breed, :description, :gender, :fixed, :house_trained, :good_with_kids, :good_with_animals, :photo_url_1, :photo_url_2, :photo_url_3)
   end
 
+  def pet_type
+    params.permit(:type)[:type]
```

Comment:
I'm curious why you used :type here instead of pet_type. Is this referring to a different :type?